### PR TITLE
Rootstock added

### PR DESCRIPTION
**Pull request by PaterSantyago**

### CODE REVIEW

File: src/Lykke.Quintessence.Rootstock.DomainServices/RootstockBlockchainService.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+            if (await _ethApiClient.GetTransactionAsync(transaction.Hash) != null)
+            {
+                return transaction.Hash;
+            }
+
+
+            var transactionHasBeenBroadcasted = false;
+            
+            for (var i = 0; i < 30; i++)
```
Comment (OttoVT) [inline src/Lykke.Quintessence.Rootstock.DomainServices/RootstockBlockchainService.cs:41]: Maybe polly could be used here for retires?

Comment (OttoVT) [reply src/Lykke.Quintessence.Rootstock.DomainServices/RootstockBlockchainService.cs:41]: I thought it is like common agreement

File: src/Lykke.Quintessence.Rootstock.DomainServices/Strategies/RootstockAddChecksumStrategy.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        {
+            _chainId = chainId;
+            _hashCalculator = hashCalculator;
+        }
+
+        
+        public async Task<string> ExecuteAsync(
+            string address)
+        {
+            address = address.Remove(0, 2).ToLowerInvariant();
```
Comment (OttoVT) [inline src/Lykke.Quintessence.Rootstock.DomainServices/Strategies/RootstockAddChecksumStrategy.cs:28]: check address to be null?

File: src/Lykke.Quintessence.Rootstock.DomainServices/Strategies/RootstockAddChecksumStrategy.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+            var addressBytes = Encoding.UTF8.GetBytes($"{_chainId.Value}0x{address}");
+            var caseMapBytes = await _hashCalculator.SumAsync(addressBytes);
+            
+            var addressBuilder = new StringBuilder("0x");
+            
+            for (var i = 0; i < 40; i++)
```
Comment (OttoVT) [inline src/Lykke.Quintessence.Rootstock.DomainServices/Strategies/RootstockAddChecksumStrategy.cs:35]: Explain this, sir. just give a link where this process is explained

**Review by OttoVT — CHANGES_REQUESTED** [review]

File: src/Lykke.Quintessence.Rootstock.DomainServices/Strategies/RootstockAddChecksumStrategy.cs (hunk 3 above)
Comment (PaterSantyago) [reply src/Lykke.Quintessence.Rootstock.DomainServices/Strategies/RootstockAddChecksumStrategy.cs:35]: See https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md for base algorithm. RSK implementation is not documented and extracted from MyCrypto source code.

File: src/Lykke.Quintessence.Rootstock.DomainServices/Strategies/RootstockAddChecksumStrategy.cs (hunk 2 above)
Comment (PaterSantyago) [reply src/Lykke.Quintessence.Rootstock.DomainServices/Strategies/RootstockAddChecksumStrategy.cs:28]: This strategy is never used with data, that we got from external sources.

File: src/Lykke.Quintessence.Rootstock.DomainServices/RootstockBlockchainService.cs (hunk 1 above)
Comment (PaterSantyago) [reply src/Lykke.Quintessence.Rootstock.DomainServices/RootstockBlockchainService.cs:41]: Big nuget for single and very simple task seems over-engineering to me

**Review by OttoVT — APPROVED** [review]

**Pull request merged**